### DISCUSSION TITLE
Is there anyway to use callback function in covalent.electron?

### DISCUSSION THREAD

**Discussion by xjf729** [AgnostiqHQ/covalent#1673] [Q&A]
If I want to do something (invoke a python function) when a covalent.electron is failure or be cancelled, what's the suggested way to implement it.

**Comment by santoshkumarradha** [comment]
Hey @xjf729 , thanks for the question.

Currently, there is no direct feature to call a function based on a change of electron status; although that would be a nice feature to have, it would be great if you could create an issue with a feature request tag and suggested UX (for instance would you like to do something in the remote machine where the execution is happening or on the place where the covalent server is running?) 

But having said that, we do have [`call_after` hooks](https://covalent.readthedocs.io/en/latest/how_to/coding/add_callable_dependencies_to_electron.html) that will let you define arbitrary functions post execution in which you can check and run various things inside the remote machine post function completion. 

PS: If you want workflows to be dynamic based on the success or failure of previous electrons, i.e. you want your next electron to do something else based on the success of the previous electron, I would suggest you return inside the electron, a state variable and then use dynamic sublattice as shown in [example here](https://covalent.readthedocs.io/en/latest/tutorials/3_QuantumChemistry/dynamic_quantum_chemistry/source.html) to build a new dynamic graph based on the status of the previous electron.

**Reply by xjf729** [reply]
Thanks for your reply. I have created an issue https://github.com/AgnostiqHQ/covalent/issues/1676 about that.
